### PR TITLE
myWidget Ad extension

### PR DESCRIPTION
**Pull request by 3lvcz**
Pull-request for "myWidget" ad.

Closes [#6965](https://github.com/ampproject/amphtml/issues/6965)

### CODE REVIEW

**Comment by 3lvcz** [comment]
I signed it!

**Comment by 3lvcz** [comment]
I signed it!

**Comment by 3lvcz** [comment]
I signed it!

File: ads/mywidget.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            unlisten();
+          }
+        });
+      });
+    },
+  };
+
+  // load the myWidget initializer asynchronously
+  loadScript(global, 'https://likemore-go.imgsmail.ru/widget.amp.js', () => {}, () => {
+    global.context.noContentAvailable();
```
Comment (lannka) [inline ads/mywidget.js]: you can make `global.context.noContentAvailable`  as the last param of `loadScript`?

File: ads/mywidget.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      cid: data.cid,
+      container: 'c',
+    },
+
+    renderStart: global.context.renderStart,
+    noContentAvailable: global.context.noContentAvailable,
+
+    ready: function(opts) {
+      // Make sure ready() is called only once.
+      if (isReady) { return; }
```
Comment (lannka) [inline ads/mywidget.js]: please break lines before `}` and after `{`

File: ads/mywidget.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      else { isReady = true; }
```
Comment (lannka) [inline ads/mywidget.js]: same here

File: ads/mywidget.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  global.myWidget = {
+    params: {
+      cid: data.cid,
+      container: 'c',
+    },
+
+    renderStart: global.context.renderStart,
+    noContentAvailable: global.context.noContentAvailable,
+
+    ready: function(opts) {
```
Comment (lannka) [inline ads/mywidget.js]: this can be simplified to:
`ready(opts) {`

File: ads/mywidget.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function mywidget(global, data) {
+  validateData(data, ['cid']);
+
+  const height = data.height;
+  let isReady = false;
+
+  if (!(height >= 0)) {
```
Comment (lannka) [inline ads/mywidget.js]: `height < 0`?

Comment (3lvcz) [inline ads/mywidget.js]: `height < 0` would not quite fit :) `data.height` can be undefined here, if attribute has invalid value or is absent, so... I can change condition to: `if (typeof height === 'undefined' || height < 0)` if you like. But there is no guarantee, that the default value of absent attribute not will be `null` tomorrow :)

Comment (lannka) [inline ads/mywidget.js]: :-) please add a little comment about this trap. to prevent "smart" people like me to refactor it later.

**Comment by 3lvcz** [comment]
I signed it!

**Comment by 3lvcz** [comment]
I recently synchronized our fork with current `master` (because our `master` revision has build errors and `amp.js` was not produced). Also I fixed all things that you mentioned in comments and tested `ads.amp.html` locally. Our Ad works fine.

But I don't understand what should I do with CLA stuff :) I already commited with my github accout one time [f3b83b8b](https://github.com/ampproject/amphtml/pull/7252/commits/f3b83b8be9536739507073dd269b9b592b88306d) and after that created an individual CLA and said "I signed it!". I missed something?

**Comment by lannka** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for the commit author(s). If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA (login here to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.

Can you try this: 1) set git email to the one signed CLA.  2) squash all the commits and then update this PR.

**Comment by 3lvcz** [comment]
I think, squash is not possible, because I pulled master between commits, so I have to squash other commits too in that case. What if we just close this pull request and create another with correct git user? Is that ok?

**Comment by 3lvcz** [comment]
Done! There is new clean pull-request: #7471

**Pull request closed without merging**